extensions/cornerstone/src/components/DicomUpload/DicomUpload.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useState } from 'react';
+import { ReactElement } from 'react';
+import Dropzone from 'react-dropzone';
+import PropTypes from 'prop-types';
+import DicomFileUploader from '../../utils/DicomFileUploader';
+import DicomUploadProgress from './DicomUploadProgress';
+import { Button } from '@ohif/ui';
+
+type DicomUploadProps = {
+  dataSource;
+  onComplete: () => void;
+  onStarted: () => void;
+};
+
+function DicomUpload({
+  dataSource,
+  onComplete,
+  onStarted,
+}: DicomUploadProps): ReactElement {
+  const [dicomFileUploaderArr, setDicomFileUploaderArr] = useState([]);
+
+  const onDrop = useCallback(async acceptedFiles => {
+    onStarted();
+    setDicomFileUploaderArr(
+      acceptedFiles.map(file => new DicomFileUploader(file, dataSource))
+    );
+  }, []);
+
+  return (
+    <div className="h-[calc(100vh-300px)] min-h-[250px] flex flex-col bg-black">
+      {dicomFileUploaderArr.length ? (
```

Comment:
I would recommend refactoring this into couple of components, or functions to make it easier to read something like 

```js
<div className="h-[calc(100vh-300px)] min-h-[250px] flex flex-col bg-black">
  {dicomFileUploaders.length ? (
    <DicomUploadProgress
      dicomFileUploaderArr={Array.from(dicomFileUploaders)}
      onComplete={onComplete}
    />
  ) : (
    renderDropzone()
  )}
</div>
```